### PR TITLE
Added ChatSDK & more complete binding for all libraries

### LINKED ISSUE #8: Big parts of SDK are not bound

### PR DESCRIPTION
**Pull request by thisisthekap**
Fixes #8 and #9

### CODE REVIEW

**Comment by thisisthekap** [comment]
@wcoder @TimofeyBurak @sleushunou @vadimpylsky Whom should I reach out to to get this PR reviewed & merged?

**Comment by thisisthekap** [comment]
@wcoder @TimofeyBurak @sleushunou @vadimpylsky Are you able to provide me with the log files of the failed CI build?

**Comment by yauhenipakala** [comment]
@thisisthekap The review is in progress and takes a lot of time. Thank you for waiting.

Could you please sync with the latest master branch to fix CI check?

**Comment by thisisthekap** [comment]
@wcoder Did the merge.

File: nuget/Softeq.ZendeskCommonUISDK.iOS.nuspec [hunk 1]
```diff
@@ -16,7 +16,9 @@
         <copyright>Copyright 2022 Softeq Development Corp.</copyright>
         <tags>zendesk, xamarin.ios, softeq, ios</tags>
         <dependencies>
-            <group targetFramework="Xamarin.iOS10" />
+            <group targetFramework="Xamarin.iOS10">
+                <dependency id="Softeq.ZendeskSDKConfigurationsSDK.iOS" version="1.1.11.4" />
```
Comment (yauhenipakala) [inline nuget/Softeq.ZendeskCommonUISDK.iOS.nuspec:20]: Why is this package required as a dependency?
As I see native [CommonUISDK](https://github.com/zendesk/commonui_sdk_ios/search?q=SDKConfigurations) doesn't use SDKConfigurations.

File: nuget/Softeq.ZendeskMessagingSDK.iOS.nuspec [hunk 2]
```diff
@@ -18,8 +18,9 @@
         <dependencies>
             <!-- Podspec: https://github.com/CocoaPods/Specs/blob/master/Specs/4/5/1/ZendeskMessagingSDK/3.8.5/ZendeskMessagingSDK.podspec.json -->
             <group targetFramework="Xamarin.iOS10">
-                <dependency id="Softeq.ZendeskCommonUISDK.iOS" version="6.1.4" />
-                <dependency id="Softeq.ZendeskMessagingAPISDK.iOS" version="3.8.5" />
+                <dependency id="Softeq.ZendeskCommonUISDK.iOS" version="6.1.4.4" />
+                <dependency id="Softeq.ZendeskMessagingAPISDK.iOS" version="3.8.5.4" />
+                <dependency id="Softeq.ZendeskSDKConfigurationsSDK.iOS" version="1.1.11.4" />
```
Comment (yauhenipakala) [inline nuget/Softeq.ZendeskMessagingSDK.iOS.nuspec:23]: This package has only 2 dependencies: https://github.com/zendesk/messaging_sdk_ios/blob/1a1645c33a00c0705cbe95d536de1b1846e4dcef/ZendeskMessagingSDK.podspec#L25-L26
As you can see ZendeskSDKConfigurationsSDK package will be added based with ZendeskMessagingAPISDK

File: nuget/Softeq.ZendeskSupportSDK.iOS.nuspec [hunk 3]
```diff
@@ -18,8 +18,12 @@
...
             <!-- Podspec: https://github.com/CocoaPods/Specs/blob/master/Specs/8/3/7/ZendeskSupportSDK/5.5.0/ZendeskSupportSDK.podspec.json -->
             <group targetFramework="Xamarin.iOS10">
-                <dependency id="Softeq.ZendeskSupportProvidersSDK.iOS" version="5.5.0" />
-                <dependency id="Softeq.ZendeskMessagingSDK.iOS" version="3.8.5" />
+                <dependency id="Softeq.ZendeskCommonUISDK.iOS" version="6.1.4.4" />
+                <dependency id="Softeq.ZendeskMessagingAPISDK.iOS" version="3.8.5.4" />
+                <dependency id="Softeq.ZendeskMessagingSDK.iOS" version="3.8.5.4" />
+                <dependency id="Softeq.ZendeskSDKConfigurationsSDK.iOS" version="1.1.11.4" />
+                <dependency id="Softeq.ZendeskSupportProvidersSDK.iOS" version="5.5.0.4" />
+                <dependency id="Softeq.ZendeskCoreSDK.iOS" version="2.7.0.4" />
```
Comment (yauhenipakala) [inline nuget/Softeq.ZendeskSupportSDK.iOS.nuspec:26]: Looks like this package has only 2 dependencies: https://github.com/zendesk/support_sdk_ios/blob/da1a8684a413206f24ebddfed35fe2499e96fdfd/ZendeskSupportSDK.podspec#L25-L26

File: SupportSDK.Bindings/SupportSDK.Bindings.csproj [hunk 4]
```diff
@@ -46,27 +46,33 @@
...
-      <Frameworks>Foundation</Frameworks>
+      <SmartLink>False</SmartLink>
+      <ForceLoad>True</ForceLoad>
     </NativeReference>
     <NativeReference Include="..\MessagingSDK.Bindings\MessagingSDK.framework">
       <Kind>Framework</Kind>
       <SmartLink>False</SmartLink>
+      <ForceLoad>True</ForceLoad>
     </NativeReference>
     <NativeReference Include="..\CommonUISDK.Bindings\CommonUISDK.framework">
```
Comment (yauhenipakala) [inline SupportSDK.Bindings/SupportSDK.Bindings.csproj:57]: Just a comment: CommonUISDK.Bindings project is not required

**Review by yauhenipakala — CHANGES_REQUESTED** [review]
Bindings work well for Chat SDK ~but don't work for Support SDK~:

~Basic `ZDKHelpCenterUi.BuildHelpCenterOverviewUi()` returns null.~
~Some classes can' t be load, for example:~
> ~System.Exception: Could not create an native instance of the type 'ZendeskCoreSDK.Bindings.ZDKObjCJwt': the native class hasn't been loaded.~

Please try these examples:
```cs
UIViewController requestList = ZDKRequestUi.BuildRequestUiWith(new NSObject[]{});
            
UIViewController helpCenter = ZDKHelpCenterUi.BuildHelpCenterOverviewUiWithConfigs(new ZDKConfiguration[]{});
            
UIViewController helpCenter = ZDKHelpCenterUi.BuildHelpCenterOverviewUi();
            
PresentModalViewController(helpCenter, true);
```

Support SDK also works fine. It was local issues with Rider and MSBuild caches.

**Review by yauhenipakala — APPROVED** [review]

**Comment by yauhenipakala** [comment]
@thisisthekap Thanks for the contribution! 🚀
New packages are available in NuGet:
https://www.nuget.org/packages?q=Tags%3A%22zendesk%22%2C%22ios%22%2C%22softeq%22

File: nuget/Softeq.ZendeskCommonUISDK.iOS.nuspec (hunk 1 above)
Comment (yauhenipakala) [reply nuget/Softeq.ZendeskCommonUISDK.iOS.nuspec:20]: Fixed in #11

File: nuget/Softeq.ZendeskMessagingSDK.iOS.nuspec (hunk 2 above)
Comment (yauhenipakala) [reply nuget/Softeq.ZendeskMessagingSDK.iOS.nuspec:23]: Fixed in #11

File: nuget/Softeq.ZendeskSupportSDK.iOS.nuspec (hunk 3 above)
Comment (yauhenipakala) [reply nuget/Softeq.ZendeskSupportSDK.iOS.nuspec:26]: Fixed in #11

**Pull request merged**